devices/src/virtio/vsock/packet.rs

```diff
@@ -342,3 +342,292 @@ impl VsockPacket {
+    }
+
+    #[test]
+    fn test_rx_packet_assembly() {
+        // Test case: successful RX packet assembly.
+        {
+            create_context!(test_ctx, handler_ctx);
+            let _pkt = VsockPacket::from_rx_virtq_head(
+                &handler_ctx.handler.rxvq.pop(&test_ctx.mem).unwrap(),
+            )
+            .unwrap();
+        }
+
+        // Test case: read-only RX packet header.
+        {
+            create_context!(test_ctx, handler_ctx);
+            handler_ctx.guest_rxvq.dtable[0].flags.set(0);
+            expect_asm_error!(rx, test_ctx, handler_ctx, VsockError::UnwritableDescriptor);
+        }
+
+        // Test case: RX descriptor head cannot fit the entire packet header.
+        {
+            create_context!(test_ctx, handler_ctx);
+            handler_ctx.guest_rxvq.dtable[0]
+                .len
+                .set(VSOCK_PKT_HDR_SIZE as u32 - 1);
+            expect_asm_error!(rx, test_ctx, handler_ctx, VsockError::HdrDescTooSmall(_));
+        }
+
+        // Test case: RX descriptor chain is missing the packet buffer descriptor.
+        {
+            create_context!(test_ctx, handler_ctx);
+            handler_ctx.guest_rxvq.dtable[0]
+                .flags
+                .set(VIRTQ_DESC_F_WRITE);
+            expect_asm_error!(rx, test_ctx, handler_ctx, VsockError::BufDescMissing);
+        }
+    }
+
+    #[test]
+    #[allow(clippy::cognitive_complexity)]
+    fn test_packet_hdr_accessors() {
+        const SRC_CID: u64 = 1;
+        const DST_CID: u64 = 2;
+        const SRC_PORT: u32 = 3;
+        const DST_PORT: u32 = 4;
+        const LEN: u32 = 5;
+        const TYPE: u16 = 6;
+        const OP: u16 = 7;
+        const FLAGS: u32 = 8;
+        const BUF_ALLOC: u32 = 9;
+        const FWD_CNT: u32 = 10;
+
+        create_context!(test_ctx, handler_ctx);
+        let mut pkt =
+            VsockPacket::from_rx_virtq_head(&handler_ctx.handler.rxvq.pop(&test_ctx.mem).unwrap())
+                .unwrap();
+
+        // Test field accessors.
+        pkt.set_src_cid(SRC_CID)
+            .set_dst_cid(DST_CID)
+            .set_src_port(SRC_PORT)
+            .set_dst_port(DST_PORT)
+            .set_len(LEN)
+            .set_type(TYPE)
+            .set_op(OP)
+            .set_flags(FLAGS)
+            .set_buf_alloc(BUF_ALLOC)
+            .set_fwd_cnt(FWD_CNT);
+
+        assert_eq!(pkt.src_cid(), SRC_CID);
+        assert_eq!(pkt.dst_cid(), DST_CID);
+        assert_eq!(pkt.src_port(), SRC_PORT);
+        assert_eq!(pkt.dst_port(), DST_PORT);
+        assert_eq!(pkt.len(), LEN);
+        assert_eq!(pkt.type_(), TYPE);
+        assert_eq!(pkt.op(), OP);
+        assert_eq!(pkt.flags(), FLAGS);
+        assert_eq!(pkt.buf_alloc(), BUF_ALLOC);
+        assert_eq!(pkt.fwd_cnt(), FWD_CNT);
+
+        // Test individual flag setting.
+        let flags = pkt.flags() | 0b1000;
+        pkt.set_flag(0b1000);
+        assert_eq!(pkt.flags(), flags);
+
+        // Test packet header as-slice access.
```

Comment:
outdated comment?

Reply:
> No; why do you say that? There are tests below, that check packet header access via the byte-slice returned by `pkt.hdr()`.

Because the tests were two lines apart from the comment and it gave the impression like you had removed something in between.